apps/backoffice-v2/src/pages/MerchantMonitoring/components/MerchantMonitoringTable/columns.tsx

```diff
@@ -160,7 +163,12 @@ export const useColumns = ({ isDemoAccount = false }) => {
         cell: ({ row }) => {
           const violations = (row.original.data?.allViolations ?? [])
             .filter(
-              el => el.id !== NO_VIOLATION_DETECTED_RISK_INDICATOR_ID && el.name && el.riskLevel,
+              (violation, index, self) =>
+                violation.id !== NO_VIOLATION_DETECTED_RISK_INDICATOR_ID &&
+                violation.name &&
+                violation.riskLevel &&
+                violation.riskLevel !== POSITIVE_RISK_LEVEL_ID &&
+                index === self.findIndex(t => t.id === violation.id),
```

Comment:
I feel like this is overcomplicated a bit, couldn't this do the same job ?
```suggestion
                self[index]?.id === violation.id,
```

Reply:
I don't think this works. You are just comparing the value at the current index to the violation ID, so all elements will always pass the filter.

If you want to remove the inner loop to lower the time complexity + make it more readable, we can maybe implement a Set to track seen violation ids?

Reply:
Ahh, so you are checking if it was seen before or not.

Then yes, I think it would be better to have a set to track such cases, or even have a filter that would pre-filter for unique values like this:
```
.filter((violation) => {
  if (seenViolations.has(violation.id)) return false;
  seenViolations.add(violation.id);
}
```

Currently for me personally its not clear that you are trying to find if this violation was displayed before, so at least creating variable like `const seenBefore = ...` would help here for code self-documentation.

But this is just a nit, maybe sometimes in the future ;-)